operator/pkg/ciliumendpointslice/controller.go

```diff
@@ -97,14 +118,23 @@ func registerController(p params) error {
 		ciliumEndpoint:      p.CiliumEndpoint,
 		ciliumEndpointSlice: p.CiliumEndpointSlice,
 		ciliumNodes:         p.CiliumNodes,
+		namespace:           p.Namespace,
 		slicingMode:         p.Cfg.CESSlicingMode,
 		maxCEPsInCES:        p.Cfg.CESMaxCEPsInCES,
 		rateLimit:           rateLimitConfig,
 		enqueuedAt:          make(map[CESName]time.Time),
 		metrics:             p.Metrics,
+		defaultCESSyncTime:  DefaultCESSyncTime,
+		priorityNamespaces:  make(map[string]struct{}),
+		Job:                 p.Job,
 	}
-
+	cesController.cond = *sync.NewCond(&lock.Mutex{})
 	p.Lifecycle.Append(cesController)
+	p.Job.Add(
```

Comment:
Should we call this here, or in the controller Start (in endpointslice.go), with the rest of the processes (e.g. "proc-queues")?

cc @ovidiutirla

Reply:
adding to job group is a safe operation, not important where we do that.

From code perspective, we are missing the init phase but it should still work fine even if we update the priority for the CES after some time (there is a chance the first reconcile will go on the standard queue until the namespace set is populated).

From readability, I think we should move this in endpointslice.go#149 (near the workerpool) and we should probably replace the workerpool with job.OneShots in a later PR.

Reply:
Sounds good.